types/application.go

```diff
@@ -20,3 +21,13 @@ type Application interface {
 	// Query for state
 	Query(query []byte) (code CodeType, result []byte, log string)
 }
+
+// Some applications can choose to implement ValidatorAware
+type ValidatorAware interface {
+
+	// Give app initial list of validators upon genesis
+	InitValidators([]*Validator)
```

Comment:
Oh uhhhhh....  what I'm thinking for that is, there's MessageType_Exception, so the TMSP server can just write a *Response of Exception type, and that signals the client (Tendermint Core) that the app crashed.